client/client.go

```diff
@@ -501,7 +501,14 @@ func (c *Client) Push(ctx context.Context, ref string, desc ocispec.Descriptor,
 		limiter = semaphore.NewWeighted(int64(pushCtx.MaxConcurrentUploadedLayers))
 	}
 
-	return remotes.PushContent(ctx, pusher, desc, c.ContentStore(), limiter, pushCtx.PlatformMatcher, wrapper)
+	// Add this handler to update layer's label when push to a new registry,
+	// After that the new registry been source too, so same layer can be mount in next push procedure
+	appendDistSrcLabelHandler, err := docker.AppendDistributionSourceLabel(c.ContentStore(), ref)
+	if err != nil {
+		return err
+	}
+
+	return remotes.PushContent(ctx, pusher, desc, c.ContentStore(), limiter, pushCtx.PlatformMatcher, wrapper, appendDistSrcLabelHandler)
```

Comment:
I don't think we need to change `PushContent` signature for that.

This could either injected with the `wrapper`, or just called separately after the `remotes.PushContent` succeeds.

FWIW, the second option is what we do in Moby (with containerd integration):

https://github.com/moby/moby/blob/1e033e2d983df0f20caa8da5320dee3c3c2d7d63/daemon/containerd/image_push.go#L211

The reason why it's IMO better to do it after the full push has succeeded is that when it's a part of a handler, the label would be set immediately for each blob that was pushed, regardless whether the full push succeded.
In this case, it could end up in a situation where blobs (like layers) were uploaded, but the registry rejected a manifest (for example because it references non-existing) blobs. Then there's no guarantee that the registry will actually persist these blobs.